### DISCUSSION TITLE
How to make the Crazyflie move slower with LPS?

### DISCUSSION THREAD

**Discussion by Fygas1** [bitcraze/discussions#1479] [Q&A] [answered]
I setup the LPS and I am now running the "swarmSequence.py" script using only 1 Crazyflie for now.

Is there anyway I can reduce the speed of Crazyflie when changing positions? I have seen `def send_full_state_setpoint` but I am not sure if this is what I'm looking for.

Also, I am trying to use both the flow deck and LPS. The drone takes off and lands using the Motion Commander and moves to setpoints with LPS. However the drone shakes when moving positions. Is this expected behavior?

Here is the code for reference:

```
def run_sequence(scf, sequence):
    try:
        cf = scf.cf
        print("Taking off!")
        #take_off(cf, sequence[0])
        with MotionCommander(scf) as mc:
            time.sleep(1.0)
            for position in sequence:
                print('Setting position {}'.format(position))
                end_time = time.time() + position[3]
                while time.time() < end_time:
                    cf.commander.send_position_setpoint(position[0],
                                                        position[1],
                                                        position[2], 0)
                    time.sleep(0.1)
            #land(cf, sequence[-1])
            print("Now Landing!")
    except Exception as e:
        print(e)
```

Thanks a lot!

**Comment by knmcguire** [comment, marked as answer]
This is the api documentation of the commander framework you are using: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/commander/

The full state setpoint will only indicate the setpoints that you want the crazyflie to be at that point in time, and PID doesn't technically support that (mellinger does). But I don't think that is what you want.

There are several ways for you to solve this:
* Reduce the max allowed velocity of the PID controller through parameters: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#posctlpid.
* Use the high level commander go-to and use a larger duration: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/high_level_commander/
* Use the highlevel position commander and set a velocity for the go_to message there: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/positioning/position_hl_commander/

The wrapper that goes around the commander class is technically the motion commander class, but unfortunately hasn't gotten a go_to option in there: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/positioning/motion_commander/. You can also consider writing something yourself with velocity setpoints (like a pid controller layer offboard but you'll probably only need the P gain)

**Reply by Fygas1** [reply]
Nice thanks a lot for your help!

**Reply by knmcguire** [reply]
forgot to give the link to the api documentation of the high level position commander so I've updated my answer above. https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/positioning/position_hl_commander/
